es-app/src/CollectionSystemManager.cpp

```diff
@@ -899,92 +850,58 @@ std::vector<std::string> CollectionSystemManager::getSystemsFromTheme()
-
-	for(auto sysIt = themeSys.begin(); sysIt != themeSys.end(); )
+	std::vector<std::string> systemsInUse = getSystemsInUse();
+	std::vector<std::string> themedSystems = getSystemsFromTheme();
+	
+	for(auto sysIt = themedSystems.begin(); sysIt != themedSystems.end(); )
 	{
 		if (std::find(systemsInUse.begin(), systemsInUse.end(), *sysIt) != systemsInUse.end())
 		{
-			sysIt = themeSys.erase(sysIt);
+			sysIt = themedSystems.erase(sysIt);
 		}
 		else
 		{
 			sysIt++;
 		}
 	}
-	return themeSys;
+	return themedSystems;
 }
 
-// returns which collection config files exist in the user folder
-std::vector<std::string> CollectionSystemManager::getCollectionsFromConfigFolder()
+std::vector<std::string> CollectionSystemManager::getSystemsInUse()
 {
-	std::vector<std::string> systems;
-	fs::path configPath = getCollectionsFolder();
-
-	if (fs::exists(configPath))
-	{
-		fs::directory_iterator end_itr; // default construction yields past-the-end
-		for (fs::directory_iterator itr(configPath); itr != end_itr; ++itr)
-		{
-			if (fs::is_regular_file(itr->status()))
-			{
-				// it's a file
-				std::string file = itr->path().string();
-				std::string filename = file.substr(configPath.string().size());
+	// get used systems in es_systems.cfg
+	std::vector<std::string> systemsInUse = getSystemsFromConfig();
+	// get folders assigned to  collections
+	std::vector<std::string> collectionThemeFolders = getCollectionThemeFolders();
+	// add them all to the list of systems in use
+	systemsInUse.insert(systemsInUse.end(), collectionThemeFolders.begin(), collectionThemeFolders.end());
 
-				// need to confirm filename matches config format
-				if (boost::algorithm::ends_with(filename, ".cfg") && boost::algorithm::starts_with(filename, "custom-") && filename != "custom-.cfg")
-				{
-					filename = filename.substr(7, filename.size()-11);
-					systems.push_back(filename);
-				}
-				else
-				{
-					LOG(LogInfo) << "Found non-collection config file in collections folder: " << filename;
-				}
-			}
-		}
-	}
-	return systems;
+	return systemsInUse;
 }
 
-// returns the theme folders for Automatic Collections (All, Favorites, Last Played) or generic Custom Collections folder
-std::vector<std::string> CollectionSystemManager::getCollectionThemeFolders(bool custom)
+
+std::vector<std::string> CollectionSystemManager::getEditableCollectionsNames()
 {
-	std::vector<std::string> systems;
-	for(std::map<std::string, CollectionSystemDecl>::iterator it = mCollectionSystemDeclsIndex.begin() ; it != mCollectionSystemDeclsIndex.end() ; it++ )
+	std::vector<std::string> colsSysDataNames;
+	for (auto collection : mCollectionSystems)
 	{
-		CollectionSystemDecl sysDecl = it->second;
-		if (sysDecl.isCustom == custom)
+		if (collection.second->decl.isEditable == true)
 		{
-			systems.push_back(sysDecl.themeFolder);
+			colsSysDataNames.push_back(collection.first);
 		}
 	}
-	return systems;
-}
+	return colsSysDataNames;
+};
```

Comment:
...do we need the comma? Is it even allowed?

Reply:
I *think* its allowed, although it does nothing. We certainly don't need it, I'll remove it.